presto-main/src/main/java/io/prestosql/operator/aggregation/TypedSet.java

```diff
@@ -43,12 +44,11 @@
     public static final DataSize MAX_FUNCTION_MEMORY = DataSize.of(4, MEGABYTE);
 
     private static final int INSTANCE_SIZE = ClassLayout.parseClass(TypedSet.class).instanceSize();
-    private static final int INT_ARRAY_LIST_INSTANCE_SIZE = ClassLayout.parseClass(IntArrayList.class).instanceSize();
     private static final float FILL_RATIO = 0.75f;
 
     private final Type elementType;
     private final Optional<MethodHandle> elementIsDistinctFrom;
-    private final IntArrayList blockPositionByHash;
+    private int[] blockPositionByHash;
```

Comment:
Int array list is backed by primitive array, but provides nicer interface and support for growing. No need to replace it

Reply:
Agreed. The only real issue is this code uses `get` instead of `getInt`

Reply:
Agreed. The only real issue is this code uses `get` instead of `getInt`

Reply:
The reasons I thought this was a good idea 
- Small amount of extra memory for wrapper `ClassLayout.parseClass(IntArrayList.class).instanceSize()`. 
- We don't use any of the fancy features from the wrapper class.
- Code is shorter and easier to read.
- Extra bounds checking on each `set` and `get` by the wrapper.
- `IntArrays.ensureCapacity` allocates a new array and then tries to preserve values from current array. In our usage we explicitly reset all the values anyway after `blockPositionByHash.size(hashCapacity)` calls so the extra work to preserve values is wasted.
- `KeyValuePairs` class has almost same code and it also avoids using the wrapper.
But I'm ok to drop the change if the above are not compelling enough reasons to replace the wrapper.

Reply:
> But I'm ok to drop the change if the above are not compelling enough reasons to replace the wrapper.

@raunaqmorarka Could you change the PR to use `getInt` instead of `get`?

Reply:
Done